orchestrator/graphql/schemas/process.py

```diff
@@ -108,7 +108,8 @@ async def subscriptions(
         subscription_ids = [str(s.subscription_id) for s in process.process_subscriptions]
         if not subscription_ids:
             return EMPTY_PAGE
-        filter_by_with_related_subscriptions = (filter_by or []) + [
-            GraphqlFilter(field="subscriptionIds", value=",".join(subscription_ids))
-        ]
-        return await resolve_subscriptions(info, filter_by_with_related_subscriptions, sort_by, first, after)
+        # filter_by_with_related_subscriptions = (filter_by or []) + [
+        #     GraphqlFilter(field="subscriptionIds", value=",".join(subscription_ids))
+        # ]
+        query_related_subscriptions = " OR ".join(f"subscription_id:{sub_id}" for sub_id in subscription_ids)
```

Comment:
this should become something that should add the user input ontop of the subscription check, so you never get unrelated subscriptions but still able to filter through those.
`(subscription_id in (id_1, id_2)) AND user input`

same for the other relation resolvers.

Reply:
Never a fan of raw SQL in the code, better to use the SQLalchemy functions.

Reply:
In the proposal section of the #288 issue, it's stated that the `filter_by` parameter should be removed:

> The GraphQL endpoint for subscriptions will have the following arguments schema:
> - `first: Int!` (Already there)
> - `after: Int!` (Already there)
> - `sortBy: [GraphqlSort!]` (Already there)
> - `query: String` (Replacement for `filterBy` and `tsv`)

Also the `OR` is not raw SQL, as it is getting passed into the [`websearch_to_tsquery`](https://www.postgresql.org/docs/current/textsearch-controls.html#id-1.5.11.6.4.11) function, which converts it to `|`—though I agree that `" OR ".join(...)` is not the ideal solution.

Maybe in the case of these sub-queries, we should use async sqlalchemy calls instead of the subscription resolver, if the query string syntax is fully replacing tsv/filterBy?

Reply:
I thought according to the Issue/proposal that the `filterBy` parameter was supposed to not be used in the graphQL query endpoint, and that the `websearch_to_tsquery` is supposed to be replacing it.

The `"OR"`s here admittedly aren't a very clean solution, but they were intended to be passed through the `websearch_to_tsquery` translation function. Maybe here though should just be SQLAlchemy orm queries (as @pboers1988 suggests) instead to get the related subscriptions...

Reply:
The point I'm trying to make is that no filtering is passed to the `resolve_subscriptions` underneath this line and it only adds filtering for the subscriptions that are inside the process.

Using sqlalchemy to get the subscriptions instead of `resolve_subscriptions` is an option but I don't have a strong opinion on that.

Reply:
I think using sqlalchemy instead of `resolve_subscriptions` is the better solution here since the query parameter is really for full-text searching which we don't support for processes. (and have no intention to IIRC)